client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx

```diff
@@ -92,11 +93,11 @@ const PlansWrapper: React.FC< Props > = ( props ) => {
 		? reduxHideFreePlan && 'plans-blog-onboarding' === plansIntent
 		: reduxHideFreePlan;
 
-	const onSelectPlan = ( selectedPlan: any ) => {
-		if ( selectedPlan ) {
+	const onSelectPlan = ( cartItems: MinimalRequestCartProduct[] | null ) => {
+		const planCartItem = getPlanCartItem( cartItems );
+		if ( planCartItem ) {
 			recordTracksEvent( 'calypso_signup_plan_select', {
-				product_slug: selectedPlan?.product_slug,
-				free_trial: selectedPlan?.free_trial,
```

Comment:
I remove the `free_trial` value because, according to tracks registration, it's not a valid event prop for `calypso_signup_plan_select`. https://github.com/Automattic/tracks-events-registration/blob/master/events/calypso_signup_plan_select.event.json

Reply:
Great finding! Thanks for digging deep into it. There is also a deprecation note from the backend, so I believe it's safe to remove: 31d84-pb.